File: pet/macos/Sources/SemaPet/Paths.swift
import Foundation

enum PetPaths {
    static let petDir: URL = {
        let home = FileManager.default.homeDirectoryForCurrentUser
        return home.appendingPathComponent(".sema/pet", isDirectory: true)
    }()

    static let assetsDir = petDir.appendingPathComponent("assets", isDirectory: true)
    static let runtimeFile = petDir.appendingPathComponent("runtime.json")
    static let configFile  = petDir.appendingPathComponent("config.json")

    static func ensureDirs() {
        for url in [petDir, assetsDir] {
            try? FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
        }
    }
}

File: pet/windows/src/paths.rs
use std::path::PathBuf;

pub const ASSETS_DIR_NAME: &str = "assets";
pub const CONFIG_FILE_NAME: &str = "config.json";
pub const RUNTIME_FILE_NAME: &str = "runtime.json";

pub struct PetPaths {
    pub home: PathBuf,
}

impl PetPaths {
    pub fn current_user() -> Self {
        Self {
            home: pet_home_dir(),
        }
    }

    pub fn assets_dir(&self) -> PathBuf {
        self.home.join(ASSETS_DIR_NAME)
    }

    pub fn config_file(&self) -> PathBuf {
        self.home.join(CONFIG_FILE_NAME)
    }

    pub fn runtime_file(&self) -> PathBuf {
        self.home.join(RUNTIME_FILE_NAME)
    }
}

fn pet_home_dir() -> PathBuf {
    std::env::var_os("USERPROFILE")
        .map(PathBuf::from)
        .unwrap_or_else(std::env::temp_dir)
        .join(".sema")
        .join("pet")
}
